### ISSUE TITLE
`InvalidRecord(InvalidInfo(TypeMismatch("Integer", "Float")))` in BCF parsing

### ISSUE THREAD

**Issue by Euphrasiologist** [zaeleus/noodles#137]
Hello again,

I'm parsing BCF's this time. The VCF opens fine. When I convert the gzipped VCF to a BCF using bcftools:

```console
bcftools view map.vcf.gz -o map.bcf
```

I can no longer parse the BCF. I pinpoint the error to when I am generating the `StringMaps` object:

```rust
let string_maps: StringMaps = raw_header.parse().unwrap();
```

Possibly this, https://github.com/zaeleus/noodles/blob/3da45c95580490df1abbc87b5be065bd1684c0f9/noodles-vcf/src/header/record/value/map/info.rs#L216.

I've bgzipped the culprit BCF.

[map.bcf.gz](https://github.com/zaeleus/noodles/files/10106490/map.bcf.gz)

**Comment by zaeleus** [comment]
Thanks for the report! The BCF string maps parser wasn't using the file format as context. The validator only supports VCF 4.3 and was erroneously enabled for your input, which is VCF 4.2. It's the BCF version of #128, actually.

This is now fixed in a406a3def6e7703796fd577fb8c1dc0503b186f5.

**Comment by Euphrasiologist** [comment]
Awesome, so fast! Thanks for fixing this 🥳
